infrastructure/src/main/java/org/corfudb/infrastructure/ResourceQuota.java

```diff
@@ -0,0 +1,58 @@
+package org.corfudb.infrastructure;
+
+import lombok.Getter;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * A quota for a resource that can be modeled with a long.
+ *
+ * Created by Maithem on 6/20/19.
+ */
+public class ResourceQuota {
+
+    @Getter
+    private final String name;
+
+    @Getter
+    private final long limit;
+
+    private final AtomicLong used = new AtomicLong();
+
+    public ResourceQuota(String name, long limit) {
+        this.name = name;
+        if (limit <= 0) {
+            throw new IllegalArgumentException("Resource limit has to be positive");
+        }
+        this.limit = limit;
+    }
+
+    /**
+     * Consumes an amount of the resource quota.
+     */
+    public void consume(long amount) {
```

Comment:
The class name might be misleading, but this is more of a tracker than "access control". 

Actually, I would like it to always consume (even it exceeds the limit), because some writes bypass the quota check and still consume some space over the limit, so we need to track that as well. Might change the class name to a ResourceTracker, if thats clearer.